### ISSUE TITLE
opens a zip-formatted comic book in a scrambled order

### ISSUE THREAD

**Issue by JohnZp** [mihonapp/mihon#596]
### Steps to reproduce

Mihon opens a zip-formatted comic book in a scrambled order

### Expected behavior

Obtain a comic with the correct sequence.

### Actual behavior

Actually, it's a scrambled one

### Mihon version

Beta 0.16.4-6595 

### Android version

Android 14

### Device

xiaomi 12s

### Other details

Mihon opens a zip-formatted comic book in a scrambled order, but when I use Komga, the sequence is correct. I'm not sure if it's a setting issue on my end or a problem with Mihon's parsing. I'm using the latest beta version. If needed, I can provide the corresponding zip-formatted comic book file.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.4](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by JohnZp** [comment]
[xt_zkgc6.txt](https://github.com/mihonapp/mihon/files/14801656/xt_zkgc6.txt)
This is an html file. I changed the suffix and can upload it.
It contains which html corresponds to which page, and then the corresponding html has the corresponding png。
![image](https://github.com/mihonapp/mihon/assets/25192317/740efe6b-f29c-4e0f-96c3-f5b826227dfb)


> it is probably using some weird numbering inside the zip should check it out, cause right now you are giving us nothing to work with, there is no information of substance in your report, millions of people are already reading zip compressed comics as CBZ is a zip file and many already use that as the default download when using Mihon
> 
> TLDR: Open the zip and actually tell us the important info of how they are named and numbered

**Comment by JohnZp** [comment]
> I knew it would be messed up, I did not expect how messed up
> 
> Your first and biggest issue is that archive files such as ZIP/CBZ are supported but the folder structure inside is not. Any folders inside the archive file are ignored
> 
> Essentially what I'm telling you is this, Mihon will not look in multiple folders, will not read HTML or CSS and will only look at image files according to their naming in the zip container

Initially, this book was in EPUB format, and Mihon could open it locally with the correct sequence. However, when I tried to access it through Komga in Mihon, I couldn't find the corresponding book. I suspected it was due to an incompatible format that Mihon couldn't recognize. So, I converted it to ZIP format using Calibre, and then I could find the book through the Komga plugin in Mihon. However, a new problem arose: the issue of the scrambled sequence.

**Comment by JohnZp** [comment]
> You'd have to go into the file and do stuff manually, you have to rename the images to a normal naming scheme manually
> 
> Right now you have scrambled the stuff by using automation

I will try it, tks
